### PR TITLE
Google App Engine Credentials.

### PR DESCRIPTION
**Pull request by anthmgoogle**
Initial implementation of AppEngineCredentials in new module because of GAE SDK dependencies.

### CODE REVIEW

File: appengine/java/com/google/auth/appengine/AppEngineCredentials.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  private final Collection<String> scopes;
+  
+  private final boolean scopesRequired;  
+  
+  public AppEngineCredentials(Collection<String> scopes) {
+    this(scopes, null);
+  }
+
+  public AppEngineCredentials(Collection<String> scopes, AppIdentityService appIdentityService) {
+    this.scopes = scopes;
```
Comment (ejona86) [inline appengine/java/com/google/auth/appengine/AppEngineCredentials.java]: Do we need to make a defensive copy?

**Comment by ejona86** [comment]
@anthmgoogle LGTM

File: appengine/java/com/google/auth/appengine/AppEngineCredentials.java (hunk 1 above)
Comment (anthmgoogle) [inline appengine/java/com/google/auth/appengine/AppEngineCredentials.java]: Done.

**Comment by anthmgoogle** [comment]
Change to make collection immutable copy as suggested.

**Comment by ejona86** [comment]
Are we interested in doing the same in [ServiceAccountCredentials](https://github.com/google/google-auth-library-java/blob/5868c19e27efa4910c09d17f95719d1d7b4f354b/oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java#L87)?

**Comment by anthmgoogle** [comment]
Yes. We did unmodifiable in the previous gen libraries, but we should probably do immutable. Bug tracking this:
https://github.com/google/google-auth-library-java/issues/17

**Pull request merged**